packages/wonder-blocks-switch/src/components/switch.tsx

```diff
@@ -0,0 +1,224 @@
+import * as React from "react";
+import {StyleSheet} from "aphrodite";
+
+import Clickable, {
+    ClickableRole,
+    ClickableState,
+} from "@khanacademy/wonder-blocks-clickable";
+import Color, {mix} from "@khanacademy/wonder-blocks-color";
+import {
+    UniqueIDProvider,
+    View,
+    addStyle,
+} from "@khanacademy/wonder-blocks-core";
+
+type Props = {
+    /**
+     * A label for the switch.
+     * If a visible label is not provided, aria-label should be used to
+     * describe the purpose of the switch.
+     */
+    ariaLabel?: string;
+    /**
+     * Whether this compoonent is checked.
+     */
+    checked: boolean;
+    /**
+     * Whether the switch is disabled. Defaults to `false`.
+     */
+    disabled?: boolean;
+    /**
+     * The unique identifier for the switch.
+     */
+    id?: string;
+    /**
+     * Function to call when the switch is clicked.
+     * @param newCheckedState
+     * @returns {unknown}
+     */
+    onChange?: (newCheckedState: boolean) => unknown;
+    /**
+     * Test ID used for e2e testing.
+     */
+    testId?: string;
+};
+
+const StyledSpan = addStyle("span");
+
+const Switch = React.forwardRef(function Switch(
+    props: Props,
+    ref: React.ForwardedRef<HTMLButtonElement>,
+) {
+    const {ariaLabel, checked, disabled = false, id, onChange, testId} = props;
+
+    const renderSwitch = (
```

Comment:
Instead of having this be a function inside of the `Switch` definition, could you break this out into its own little component (still within this file) outside of `Switch`? This would help with optimization and be more in line with best practices, which you can read about [here](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/no-unstable-nested-components.md).